MapleServer2/AIScripts/PrisonGovernor2Battle.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Maple2Storage.Types;
+using MapleServer2.Data.Static;
+using MapleServer2.Managers.Actors;
+using MapleServer2.Types;
+
+namespace MapleServer2.AIScripts;
+
+public class PrisonGovernor2Battle : AIState
+{
+    public override void OnEnter(Npc npc)
+    {
+        // Ok to assume this will only happen on tutorial, so always attack the only player available
+        npc.Target = npc.FieldManager!.State.Players.FirstOrDefault().Value;
+    }
+
+    public override AIState? Execute(Npc npc)
+    {
+        if (npc.CheckHPThreshold(50))
+        {
+            return new BattleStop();
+        }
```

Comment:
Likewise with this. `CheckHPThreshold` would be an AI function and not necessarily under NPC, like how we have `TriggerCondition` all in one area specifically for trigger scripts